airflow/providers/apache/beam/operators/beam.py

```diff
@@ -57,8 +57,8 @@ def _set_dataflow(
 
     def __set_dataflow_hook(self) -> DataflowHook:
         self.dataflow_hook = DataflowHook(
-            gcp_conn_id=self.dataflow_config.gcp_conn_id or self.gcp_conn_id,
-            delegate_to=self.dataflow_config.delegate_to or self.delegate_to,
+            gcp_conn_id=self.dataflow_config.gcp_conn_id or self.gcp_conn_id,  # type: ignore[attr-defined]
+            delegate_to=self.dataflow_config.delegate_to or self.delegate_to,  # type: ignore[attr-defined]
             poll_sleep=self.dataflow_config.poll_sleep,
```

Comment:
In this place, I have to use ignore.
I assumed that using "Protocol" from `typing` will make the code overcomplicated. But if you insist I'll implement `Protocol`.

Reply:
I think you can fix this by adding

```python
gcp_conn_id: str
delegate_to: Optional[str]
```

to the class, under `dataflow_config: DataflowConfiguration`.

Reply:
That's it. Thanks